base_layer/wallet/src/output_manager_service/storage/sqlite_db/output_sql.rs

```diff
@@ -807,28 +807,32 @@ impl Encryptable<XChaCha20Poly1305> for OutputSql {
             .to_vec()
     }
 
-    fn encrypt(&mut self, cipher: &XChaCha20Poly1305) -> Result<(), String> {
-        self.spending_key = encrypt_bytes_integral_nonce(
+    fn encrypt(self, cipher: &XChaCha20Poly1305) -> Result<Self, String> {
+        let mut output = self.clone();
```

Comment:
Why clone self here?
Its moved in here, not borrowed. 
We clone it, leave the original to drop at the end of the function, and we don't zeroize there data. 
Before we replaced the sensitive keys with encrypted values, so we did not have to zeroize.

Reply:
Is it because we changed the trait to not be `&mut`?
If so I undo that change, as its going to lead to a lot of cases like this.

Reply:
The main purpose of the current PR was to change the design of 'encryption' and 'decryption' methods, to consume `self` instead of reusing it via `&mut self`. I have refactored this function to zeroize the sensitive data, that is encrypted, of `self`. Accordingly, 

```rs
fn encrypt(mut self, cipher: &XChaCha20Poly1305) -> Result<Self, String> {
        let mut output = self.clone();
        output.private_key =
            encrypt_bytes_integral_nonce(cipher, self.domain("private_key"), Hidden::hide(output.private_key))?;

        // zeroize sensitive data
        self.private_key.zeroize();

        Ok(output)
    }
```

Reply:
I am pretty sure you can change to those to this:
```
n encrypt(mut self, cipher: &XChaCha20Poly1305) -> Result<Self, String> {
        self.private_key =
            encrypt_bytes_integral_nonce(cipher, self.domain("private_key"), Hidden::hide(output.private_key))?;

        Ok(self)
    }
```

Then we move it in and out, we don't have to clone, we don't have to zerozie